### PR TITLE
Wire in the device observatory port while initializing the Dart VM

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: sky/shell/shell.cc [hunk 1]
```diff
@@ -81,9 +82,22 @@ void Shell::InitStandalone() {
...
       !command_line.HasSwitch(switches::kNonInteractive);
+  // Set Observatory Port
+  if (command_line.HasSwitch(switches::kDeviceObservatoryPort)) {
+    auto port_string =
+        command_line.GetSwitchValueASCII(switches::kDeviceObservatoryPort);
+    std::stringstream stream(port_string);
+    uint32_t port = 0;
+    if (stream >> port) {
+      settings.observatory_port = port;
+    }
```
Comment (krisgiesing) [inline sky/shell/shell.cc]: Maybe emit a warning if the port is malformed?

**Comment by krisgiesing** [comment]
LGTM

File: sky/shell/shell.cc (hunk 1 above)
Comment (chinmaygarde) [inline sky/shell/shell.cc]: Ok, If the switch is specified and is malformed, I log to the console.

File: sky/shell/shell.cc [hunk 2]
```diff
@@ -81,9 +82,25 @@ void Shell::InitStandalone() {
...
+  // Set Observatory Port
+  if (command_line.HasSwitch(switches::kDeviceObservatoryPort)) {
+    auto port_string =
+        command_line.GetSwitchValueASCII(switches::kDeviceObservatoryPort);
+    std::stringstream stream(port_string);
+    uint32_t port = 0;
+    if (stream >> port) {
+      settings.observatory_port = port;
+    } else {
+      LOG(INFO)
```
Comment (eseidelGoogle) [inline sky/shell/shell.cc]: Seems like a WARN or ERROR?

File: sky/shell/switches.cc [hunk 3]
```diff
@@ -26,6 +27,7 @@ void PrintUsage(const std::string& executable_name) {
             << " --" << kTraceStartup
             << " --" << kFLX << "=FLX"
             << " --" << kPackageRoot << "=PACKAGE_ROOT"
+            << " --" << kDeviceObservatoryPort << "=8181"
```
Comment (eseidelGoogle) [inline sky/shell/switches.cc]: 8181?  do you mean PORT?  Do we need to document that 0 is special?

Comment (chinmaygarde) [inline sky/shell/switches.cc]: I just wanted to specify that 8181 was the default. I can update it to say PORT

**Comment by chinmaygarde** [comment]
> use base::StringToInt instead of iostreams?

I didn't that was a thing. Do we have a preference? I can update it either way.

**Comment by jason-simmons** [comment]
Google and Chromium styles typically do not use iostreams except for logs.  I don't know if Flutter has an established preference.

**Pull request merged**